Server/AuthService/AuthService.Infrastructure/HttpClients/UserHttpClient.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using ModularHouse.Server.AuthService.Application.Dto;
+using ModularHouse.Server.AuthService.Application.HttpClients.UMS;
+using ModularHouse.Server.AuthService.Infrastructure.MappingExtensions;
+using ModularHouse.Server.Common.Domain;
+using ModularHouse.Server.Common.Domain.Exceptions;
+using ModularHouse.Shared.Models.Responses.UMS;
+
+namespace ModularHouse.Server.AuthService.Infrastructure.HttpClients;
+
+public class UserHttpClient : IUserHttpClient
+{
+    private const string BaseUri = "api/users";
+    
+    private readonly HttpClient _httpClient;
+    private readonly ILogger<UserHttpClient> _logger;
+    
+    public UserHttpClient(IOptions<HttpClientsOptions> httpClientsOptions, ILogger<UserHttpClient> logger)
+    {
+        _httpClient = new HttpClient { BaseAddress = new Uri(httpClientsOptions.Value.UMS.BaseUrl) };
+        _logger = logger;
+    }
+    
+    public async Task<UserDto> GetUserByIdAsync(Guid id, CancellationToken cancellationToken = default)
+    {
+        var requestUri = $"{BaseUri}/{id}";
+        var response = await _httpClient.GetAsync(requestUri, cancellationToken);
+        var responseBody = await response.Content.ReadAsStringAsync(cancellationToken);
+
+        if (response.IsSuccessStatusCode)
```

Comment:
Here your condition is: response.IsSuccessStatusCode, but below is: response.StatusCode != HttpStatusCodes.OK. I think it will be great to follow one code style. Also in the first condition:
if response.IsSuccessStatusCode you throw an exception. Is this the right behaviour ?